src/main/java/org/prebid/server/bidder/beachfront/BeachfrontBidder.java

```diff
@@ -499,42 +443,66 @@ private static Bid updateVideoBid(Bid bid, BeachfrontVideoImp imp, String id) {
     /**
      * Creates {@link Bid} from {@link BeachfrontResponseSlot}.
      */
-    private static Bid makeBidFromBeachfrontSlot(BeachfrontResponseSlot beachfrontResponseSlot, String id) {
-        final String adm = beachfrontResponseSlot.getAdm();
-        final Float price = beachfrontResponseSlot.getPrice();
-
+    private static Bid makeBidFromBeachfrontSlot(BeachfrontResponseSlot beachfrontResponseSlot) {
+        final String slot = beachfrontResponseSlot.getSlot();
         return Bid.builder()
-                .crid(adm != null ? extractCrid(adm, 1) : null)
-                .impid(beachfrontResponseSlot.getSlot())
-                .price(price != null ? BigDecimal.valueOf(price) : null)
-                .id(id)
-                .adm(adm)
+                .crid(beachfrontResponseSlot.getCrid())
+                .impid(slot)
+                .price(BigDecimal.valueOf(beachfrontResponseSlot.getPrice()))
+                .id(slot + "Banner")
+                .adm(beachfrontResponseSlot.getAdm())
                 .h(beachfrontResponseSlot.getH())
                 .w(beachfrontResponseSlot.getW())
                 .build();
     }
 
-    /**
-     * Extracts crid from " separated source by position. In case when crid was not found return null.
-     */
-    private static String extractCrid(String source, int position) {
-        String[] crids = source.split("\"");
-        return position < crids.length ? crids[position] : null;
+    private Result<List<BidderBid>> processVideoResponse(String responseBody, HttpRequest httpRequest) {
+        final BidResponse bidResponse = Json.decodeValue(responseBody, BidResponse.class);
+        final BeachfrontVideoRequest videoRequest = Json.decodeValue(httpRequest.getBody(),
+                BeachfrontVideoRequest.class);
+
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            return Result.of(Collections.emptyList(), Collections.emptyList());
+        }
+
+        final List<Bid> bids = bidResponse.getSeatbid().get(0).getBid();
+        final List<Imp> imps = videoRequest.getRequest().getImp();
+        if (httpRequest.getUri().contains(NURL_VIDEO_ENDPOINT_SUFFIX)) {
+            return Result.of(updateVideoBids(bids, imps).stream()
+                            .map(bid -> BidderBid.of(bid, BidType.video, DEFAULT_BID_CURRENCY))
```

Comment:
They also use `TypedBid.BidVideo` which we couldn't have.

Reply:
I know, I noticed that object, but we simply don't have one so I've left it as it is. Dunno how crucial is this, need @rpanchyk to take a look.

Reply:
that's ok, `BidType.video` is fine.